library/Exceptions/NestedValidationException.php

```diff
@@ -137,11 +137,12 @@ public function getIterator()
         $knownDepths = [];
         foreach ($recursiveIteratorIterator as $childException) {
             if ($childException instanceof self
-                && $childException->getRelated()->count() > 0
-                && $childException->getRelated()->count() < 2) {
-                continue;
+                && $childException->getRelated()->count() === 1) {
+                $relatedException = iterator_to_array($childException->getRelated())[0];
+                if (! $relatedException instanceof  AttributeException) {
```

Comment:
Hi, thanks for reviewing!

`AbstractRelated` is a `Validatable` type, not an `Exception`, so they cannot be interchanged here. There isn't a `AbstractRelatedException` - `AttributeException` is derived from `NestedValidationException`.

I think the closest equivalent to checking for `AbstractRelate` is by checking if the related exception has a `validator` param, but that doesn't seem like a very reliable / accurate check.

We could add an `AbstractRelatedException` class and make all `AbstractRelated` rules' exceptions inherit from it, but we'd run into another problem:
- These types inherit `AbstractRelated`: `Attribute`, `Call`, `Key`, `KeyNested`
- `Key` and `KeyNested` inherit `AttributeException` which inherits `NestedValidationException`
- But, `CallException` inherits `GroupedValidationException`

So, I think we'd want to make all of those rule / exception types to have congruent type ancestry.

What do you think?